flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/sink/StreamRecordTimestampInserter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.operators.sink;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.runtime.operators.TableStreamOperator;
+
+/**
+ * Operator which sets the timestamp on the StreamRecord from the corresponding column of each row.
+ */
+@Internal
+public class StreamRecordTimestampInserter extends TableStreamOperator<RowData>
+        implements OneInputStreamOperator<RowData, RowData> {
+
+    private final int rowtimeIndex;
+    private final int precision;
+
+    public StreamRecordTimestampInserter(int rowtimeIndex, int precision) {
+        this.rowtimeIndex = rowtimeIndex;
+        this.precision = precision;
+    }
+
+    public StreamRecordTimestampInserter(int rowtimeIndex) {
+        this(rowtimeIndex, 3);
+    }
+
+    @Override
+    public void processElement(StreamRecord<RowData> element) throws Exception {
+        final RowData rowData = element.getValue();
+        if (rowtimeIndex != -1) {
```

Comment:
should not be -1 at this point?

Reply:
Indeed no, I thought to keep it as an extra "safety net" but can be confusing, I'll remove.